Warzone2100/warzone2100
# Split iIMDShape, separate game state data from display data, + initial graphics overrides support

Requires: #3307

This draft PR implements several key core changes:
1. Split `iIMDShape`, separate game state data from display data
   - iIMDBaseShape is the base data that can be safely used for game state calculations, and is always derived from the base model data. It contains a pointer to the *display* iIMDShape data (used for display purposes only), which may be loaded from a different place (i.e. graphics overrides).
2. Support tileset TEXTURE overrides in PIE 4
3. Basic under-the-hood support for new "graphics overrides" support
4. Refactoring of how textures are compressed as part of the build process

## Graphics Overrides

This PR implements the basics to support graphics overrides.

The goal: Use the same (base) game data for all game state calculations / the simulation, but support using a different set of data for rendering / display purposes.

Use-case: To allow future bundling of ARMod / high quality models, and allow players to play versus each other while independently deciding whether to use classic, normal, or ARMod/HQ models for the visual appearance of the game.

To test this right now:
- Extract ARMod, and place in `<config dir>/graphics_overrides/curr` (this may change before this PR is ultimately ready for merging)
   - This should cause ARMod models and textures to be rendered, however all game state calculations should use data extracted from the base (default) models.

### Future TODOs (in a subsequent PR):

- Similar to terrain overrides, provide configuration in Options (and enumeration of graphics_overrides). (Presumably allow these to be packaged, like terrain overrides?)
